src/Refactoring-Core/RBCondition.class.st

```diff
@@ -302,9 +332,11 @@ RBCondition class >> isSymbol: aString [
 { #category : 'instance creation' }
 RBCondition class >> isValidClassName: aString [
 
-	^ self new
-		  block: [ self checkClassVarName: aString in: self ]
-		  errorString: aString , ' is <1?:not >a valid class name'
+	| validator |
+	validator := self new.
+	^ validator
+		  block: [ self checkClassVarName: aString in: self validationContext: validator ]
+		  errorString: validator errorString  
```

Comment:
I'm not sure I understand why we need `validationContext` here? Could you please explain why the old way of invoking the precondition didn't work (what it lacked)?